handbook/python-dataplugins/python-dataplugins.md

```diff
@@ -0,0 +1,308 @@
+```
+
+The class name cannot be changed.
+
+### Store read
+
+Every Python DataPlugin needs to implement a `read_store` method. This method is called by DIAdem, LabVIEW or SystemLink DataFinder when it attempts to open your data file. The applications pass a set of useful parameters that can be accessed by the parameter array.
+
+<!-- markdownlint-disable -->
+<details>
+<summary>Example Code</summary>
+<!-- markdownlint-enable -->
+
+```python
+import datetime
+import os
+from pathlib import Path
+def read_store(self, parameter):
+   """
+      Read data file and return a Python dictionary
+      that contains groups and channels in a TDM-like structure.
+   """
+
+   # String: Contains the absolute path to the data file
+   file_path = os.path.realpath(parameter["file"])
+   # Boolean: Denotes if data file was accessed by SystemLink DataFinder
+   # => the bulk data was not touched. 
+   is_datafinder_indexer = parameter["datafinder"]
+
+   tdm_tree = {
+      "author": "HelloWorkd test",
+      "description": "File containing a json dict read by Python plugin",
+      "groups": [{
+            "name": "Group_1",
+            "description": "The first group",
+            "channels": [{
+               "name": "Index",
+               "description": "",
+               "info": "Going up",
+               "unit_string": "s",
+               "type": "DataTypeChnFloat64",
+               "values": [1, 2, 3]
+            }, {
+               "name": "Vals_1",
+               "description": "",
+               "unit_string": "km/h",
+               "type": "DataTypeChnFloat64",
+               "values": [1.1, 2.1, 3.1]
+            }, {
+               "name": "Vals_2",
+               "description": "",
+               "unit_string": "km/h",
+               "type": "DataTypeChnFloat64",
+               "values": [1.2, 2.2, 3.2]
+            }, {
+               "name": "Str_1",
+               "description": "",
+               "type": "DataTypeChnString",
+               "values": ["abc", "def", "hij"]
+            }]
+      }, {
+            "name": "Group_2",
+            "description": "The first group",
+            "channels": [{
+               "name": "Index",
+               "description": "",
+               "info": "Going up",
+               "unit_string": "s",
+               "type": "DataTypeChnFloat64",
+               "values": [1, 2, 3, 4]
+            }
+            ]
+      }]
+   }
+
+   return {Path(file_path).stem: tdm_tree}
+```
+
+</details>
+
+Use the file parameter to access your file using text, CSV, or binary readers.
+The data must be added to a Python dictionary. It represents the
+[structure of tdm/tdms files](https://www.ni.com/en-us/support/documentation/supplemental/06/the-ni-tdms-file-format.html)
+that consist of one root, 0...m groups and 0...n channels:
+
+<figure>
+   <img src="../../img/pydp-tdm_structure.png" width="500" />
+   <figcaption>TDM structure with file, groups and channels</figcaption>

```

Comment:
I think we need to decide whether to use the term "root" or "file" for the top level.

Reply:
yes root is probably better here